iroh-bytes/src/provider.rs

```diff
@@ -487,34 +488,50 @@ pub enum SentStatus {
     NotFound,
 }
 
-/// Send a
+/// Send a blob to the client.
 pub async fn send_blob<D: Map, W: AsyncStreamWriter>(
     db: &D,
-    name: Hash,
+    hash: Hash,
     ranges: &RangeSpec,
     mut writer: W,
 ) -> Result<(SentStatus, u64, SliceReaderStats)> {
-    match db.get(&name).await? {
+    match db.get(&hash).await? {
         Some(entry) => {
             let outboard = entry.outboard().await?;
             let size = outboard.tree().size();
             let mut file_reader = TrackingSliceReader::new(entry.data_reader().await?);
             writer.write(size.to_le_bytes().as_slice()).await?;
-            let res = encode_ranges_validated(
+            encode_ranges_validated(
                 &mut file_reader,
                 outboard,
                 &ranges.to_chunk_ranges(),
                 writer,
             )
-            .await;
-            debug!("done sending blob {} {:?}", name, res);
-            res?;
+            .await
+            .map_err(|e| encode_error_to_anyhow(e, &hash))?;
 
             Ok((SentStatus::Sent, size, file_reader.stats()))
         }
         _ => {
-            debug!("blob not found {}", hex::encode(name));
+            debug!("blob not found {}", hash.to_hex());
             Ok((SentStatus::NotFound, 0, SliceReaderStats::default()))
         }
     }
 }
+
+fn encode_error_to_anyhow(err: EncodeError, hash: &Hash) -> anyhow::Error {
```

Comment:
Because anyhow makes it particularly hard to downcast the hash (and rightly so, you probably need thiserror if that's what you want to support) another approach - which I usually take - is to add a tracing span with the hash because it's purely a reporting thing now.  You can do an `#[instrument(...)]` to `send_blob` but probably also want to start the span even before this.